test/unit/adapters/ws-pool.spec.ts

```diff
@@ -0,0 +1,209 @@
+import { createWSPoolComponent } from '../../../src/adapters/ws-pool'
+import { mockConfig, mockMetrics, mockRedis } from '../../mocks/components'
+import { IWSPoolComponent } from '../../../src/types'
+
+describe('ws-pool-component', () => {
+  let wsPool: IWSPoolComponent
+  let mockRedisClient: jest.Mocked<any>
+  let originalSetInterval: typeof setInterval
+  let mockSetInterval: jest.Mock
+  let mockClearInterval: jest.Mock
+
+  beforeEach(async () => {
+    jest.clearAllMocks()
+
+    // Mock setInterval/clearInterval
+    originalSetInterval = global.setInterval
+    mockSetInterval = jest.fn()
+    mockClearInterval = jest.fn()
+    global.setInterval = mockSetInterval as any
+    global.clearInterval = mockClearInterval
+
+    mockRedisClient = mockRedis.client as jest.Mocked<any>
+    mockRedisClient.zCard.mockResolvedValue(0)
+    mockRedisClient.exists.mockResolvedValue(0)
+    mockRedisClient.set.mockResolvedValue('OK')
+    mockRedisClient.zAdd.mockResolvedValue(1)
+    mockRedisClient.multi.mockReturnValue({
+      set: jest.fn().mockReturnThis(),
+      zAdd: jest.fn().mockReturnThis(),
+      exec: jest.fn().mockResolvedValue(['OK', 1])
+    })
+
+    // Mock redis.get and redis.put for the component
+    mockRedis.get.mockResolvedValue(null)
+    mockRedis.put.mockResolvedValue(undefined)
+
+    mockConfig.getNumber.mockImplementation(async (key) => {
+      if (key === 'IDLE_TIMEOUT_IN_MS') return 300000
+      return 0
+    })
+
+    wsPool = await createWSPoolComponent({ metrics: mockMetrics, config: mockConfig, redis: mockRedis })
+  })
+
+  afterEach(() => {
+    global.setInterval = originalSetInterval
+  })
+
+  describe('initialization', () => {
+    it('should initialize with default idle timeout if config returns falsy', async () => {
+      mockConfig.getNumber.mockResolvedValueOnce(0)
+      const pool = await createWSPoolComponent({ metrics: mockMetrics, config: mockConfig, redis: mockRedis })
+      expect(pool).toBeDefined()
+    })
+
+    it('should use configured idle timeout', async () => {
+      const customTimeout = 60000
+      mockConfig.getNumber.mockImplementation(async (key) => (key === 'IDLE_TIMEOUT_IN_MS' ? customTimeout : 0))
+      const pool = await createWSPoolComponent({ metrics: mockMetrics, config: mockConfig, redis: mockRedis })
+      expect(pool).toBeDefined()
+    })
```

Comment:
I'm not sure if we really want to put efforts on testing this 😆 but if we do really want to, should we add an `expect(redisMock.set).toHaveBeenCalledWith({...})` after calling `acquireConnection`?